tests/integration/helpers.py

```diff
@@ -145,3 +161,188 @@ async def wait_for_mongos_units_blocked(
                 await check_all_units_blocked_with_status(ops_test, db_app_name, status)
     finally:
         await ops_test.model.set_config({hook_interval_key: old_interval})
+
+
+async def deploy_cluster_components(ops_test: OpsTest) -> None:
+    """Deploys all cluster components and waits for idle."""
+    mongos_charm = await ops_test.build_charm(".")
+    resources = {
+        "mongodb-image": METADATA["resources"]["mongodb-image"]["upstream-source"]
+    }
+    await ops_test.model.deploy(
+        mongos_charm,
+        resources=resources,
+        application_name=MONGOS_APP_NAME,
+        series="jammy",
+    )
+
+    await ops_test.model.deploy(
+        MONGODB_CHARM_NAME,
+        application_name=CONFIG_SERVER_APP_NAME,
+        channel="6/edge",
+        resources=resources,
+        config={"role": "config-server"},
+    )
+    await ops_test.model.deploy(
+        MONGODB_CHARM_NAME,
+        application_name=SHARD_APP_NAME,
+        channel="6/edge",
+        resources=resources,
+        config={"role": "shard"},
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[MONGOS_APP_NAME, SHARD_APP_NAME, CONFIG_SERVER_APP_NAME],
+        idle_period=10,
+        raise_on_blocked=False,
```

Comment:
Because of DPE-4996 we should also `raise_on_error=False` here

Reply:
Ah yes thank you!